pyvista/core/_validation/validate.py

```diff
@@ -521,7 +521,6 @@ def validate_transform4x4(transform, /, *, name="Transform"):
                 arr = validate_array(
                     transform,
                     must_have_shape=(4, 4),
-                    must_be_finite=True,
```

Comment:
This removal is needed because this test:
https://github.com/pyvista/pyvista/blob/2930f1f7ffafb454e99b3aa51714ea9a82ecd1b9/tests/core/test_dataset.py#L281
Checks that inputs with `inf` and `nan` values do _not_ raise any errors. This is done since the vtk methods support `inf` and `nan` inputs. But is this what we want? Are there real-world use cases where `nan` and `inf` are desirable to have in a transformation matrix?
If not I think we should keep `must_be_finite=True` here and modify the test instead to _not_ allow `inf` and `nan` in a transformation.

Reply:
This should be addressed as part of #6484.